test/functional/interface_usdt_net.py

```diff
@@ -77,12 +112,112 @@
     outbound_messages.perf_submit(ctx, &msg, sizeof(msg));
     return 0;
 };
+
+BPF_PERF_OUTPUT(inbound_connections);
+int trace_inbound_connection(struct pt_regs *ctx) {
+    struct NewConnection inbound = {};
+    bpf_usdt_readarg(1, ctx, &inbound.conn.id);
+    bpf_usdt_readarg_p(2, ctx, &inbound.conn.addr, MAX_PEER_ADDR_LENGTH);
```

Comment:
> What would it do if the argument is longer than the supplied buffer? I presume the last arg is the length of the buffer, so it is not going to overflow, I guess. 

Correct, it would cut it off.

> But would it add a trailing '\0'?

No. However there is [bpf_probe_read_user_str()](https://docs.ebpf.io/linux/helper-function/bpf_probe_read_user_str/) which does that. This generally seems to be the better suited methodology here. I'll use that for the changes in this PR. There are other places in our code we can update in a potential follow up.

> That is - could it copy MAX_PEER_ADDR_LENGTH from the input to the output and leave it like that? If yes, would that be a problem - a string without a terminating '\0'?

I think, for us, in this test here, this isn't a problem. The Python ctype mapping for `Connection` defined [below](https://github.com/bitcoin/bitcoin/pull/25832/commits/3c16937f66523e92eb81afacff285fdab80e2e48#diff-e4179f6d1b7cda2cccb1b611452a520817d2d94a8c3f86f00ede8e3dd03b0b61R113-R119) has the same size for an address (`("addr", ctypes.c_char * MAX_PEER_ADDR_LENGTH)`).